core/src/server/rpc_module.rs

```diff
@@ -884,17 +895,20 @@ impl Drop for SubscriptionSink {
 /// Wrapper struct that maintains a subscription "mainly" for testing.
 #[derive(Debug)]
 pub struct Subscription {
-	tx: async_channel::Sender<()>,
+	close_notify: Option<Arc<Notify>>,
 	rx: mpsc::UnboundedReceiver<String>,
 	sub_id: RpcSubscriptionId<'static>,
 }
 
 impl Subscription {
 	/// Close the subscription channel.
 	pub fn close(&mut self) {
-		self.tx.close();
+		tracing::trace!("[Subscription::close] Notifying");
+		self.close_notify.take().and_then(|n| {
```

Comment:
I think this is cleaner

```suggestion
if let Some(n) = self.close_notify.take() { 
     n.notify_one() 
}
```